src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
```

Comment:
You should use generics here,
```suggestion
    private T[] elementData;
```